internal/handlers/pg/msg_listcollections.go

```diff
@@ -93,20 +90,39 @@ func (h *Handler) MsgListCollections(ctx context.Context, msg *wire.OpMsg) (*wir
 			continue
 		}
 
+		if nameOnly {
+			d = must.NotFail(types.NewDocument(
+				"name", n,
+			))
+		}
+
 		collections.Append(d)
 	}
 
 	var reply wire.OpMsg
-	must.NoError(reply.SetSections(wire.OpMsgSection{
-		Documents: []*types.Document{must.NotFail(types.NewDocument(
-			"cursor", must.NotFail(types.NewDocument(
-				"id", int64(0),
-				"ns", db+".$cmd.listCollections",
-				"firstBatch", collections,
-			)),
-			"ok", float64(1),
-		))},
-	}))
+
+	switch {
+	case nameOnly:
+		must.NoError(reply.SetSections(wire.OpMsgSection{
+			Documents: []*types.Document{must.NotFail(types.NewDocument(
+				"cursor", must.NotFail(types.NewDocument(
+					"firstBatch", collections,
+				)),
+				"ok", float64(1),
+			))},
+		}))
+	default:
```

Comment:
Yes, it seems that mongodb returns those fields:
```
test> db.runCommand({listCollections:1,nameOnly:true})
{
  cursor: {
    id: Long("0"),
    ns: 'test.$cmd.listCollections',
    firstBatch: [ { name: 'values', type: 'collection' } ]
  },
  ok: 1
}
```

Reply:
Hi @AlekSi , @noisersup 
I added `id` and `ns`.
I kept `switch` statement, because when `nameOnly: false` is set it returns the output with more information which would be good to be add later.

```
db.runCommand( { listCollections: 1, nameOnly: false } )
{
  cursor: {
    id: Long("0"),
    ns: 'admin.$cmd.listCollections',
    firstBatch: [
      {
        name: 'system.users',
        type: 'collection',
        options: {},
        info: {
          readOnly: false,
          uuid: new UUID("9b765d1e-a94f-4b0a-b28e-727ce544c071")
        },
        idIndex: { v: 2, key: { _id: 1 }, name: '_id_' }
      },
      {
        name: 'system.version',
        type: 'collection',
        options: {},
        info: {
          readOnly: false,
          uuid: new UUID("f5b30138-9350-4075-8d91-a9886dd0eb3f")
        },
        idIndex: { v: 2, key: { _id: 1 }, name: '_id_' }
      }
    ]
  },
  ok: 1
}
```

Thanks!

Reply:
> I kept switch statement, because when nameOnly: false is set it returns the output with more information which would be good to add later.

So we would add a switch later, too :) Right now, that code looks like a result of a mis-merge or copy&paste error. Let's simplify it.

Reply:
@AlekSi updated